From: Josselin Mouette <joss@debian.org>
Date: Wed, 28 Jul 2010 11:59:24 +0200
Subject: [PATCH] =?UTF-8?q?Don=E2=80=99t=20blindly=20listen=20to=20any=20signal?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

that goes through the session bus. Gnome-session will know when to kill
the applet.

Closes: #486674
Thanks:  Michal Čihař
---
 applet.py |   15 ---------------
 1 files changed, 0 insertions(+), 15 deletions(-)

diff --git a/applet.py b/applet.py
index 491ddb1..28f681a 100644
--- a/applet.py
+++ b/applet.py
@@ -329,21 +329,6 @@ if __name__ == '__main__':
         except:
             pass

-    if trayicon:
-        # Stop running when the session ends.
-        def monitor_session (*args):
-            pass
-
-        try:
-            bus = dbus.SessionBus()
-            bus.add_signal_receiver (monitor_session)
-        except:
-            try:
-                print >> sys.stderr, ("%s: failed to connect to "
-                                      "session D-Bus" % PROGRAM_NAME)
-            finally:
-                sys.exit (1)
-
     try:
         bus = dbus.SystemBus()
     except:
--
